backend/parser.py

```diff
@@ -49,42 +68,140 @@ def add_point(self, timestamp, data: list, latency=None):
         else:
             self.timestamp_diffs.append(0)
 
-        if latency != None:
+        if latency >= 0:
             self.latencies.append(latency)
 
-        for i, point in enumerate(data):
+        for i, point in enumerate(datapoints):
             self.data[i].append(point)
             
         self.next_index += 1
 
+    def set_dimensions(self, dimensions: int):
+        """Initializes empty lists in dict after parser determines data dimensions.
+        
+        Args:
+            dimensions: The number of dimensions that the data has. e.g. if the
+                gyroscope returns (x,y,z) tuples the parser calls set_dimensions(3).
+        
+        Raises:
+            ValueError: "< 1 dimension error". 0 or negative dimensions are not valid.
+        """
+
+        if dimensions < 1:
+            raise ValueError("< 1 dimension error")
+        if dimensions == 1:
+            self.performed_dimension_set = True
```

Comment:
This variable seems unused to me

Reply:
performed_dimensions_set is going to be used when reading data lines from a file. I haven't pushed the code yet but the parser will use this flag to automatically determine a sensors data dimensions when reading the file.